api/src/main/java/org/openmrs/module/registrationcore/api/impl/PatientActionListener.java

```diff
@@ -0,0 +1,72 @@
+package org.openmrs.module.registrationcore.api.impl;
+
+import org.openmrs.Patient;
+import org.openmrs.api.APIException;
+import org.openmrs.api.PatientService;
+import org.openmrs.api.context.Context;
+import org.openmrs.event.EventListener;
+import org.openmrs.module.registrationcore.RegistrationCoreConstants;
+
+import javax.jms.JMSException;
+import javax.jms.MapMessage;
+import javax.jms.Message;
+
+public abstract class PatientActionListener implements EventListener {
```

Comment:
Just from peeking at https://wiki.openmrs.org/display/docs/Event+Module I wonder if we should have extended SubscribableEventListener. (But haven't looked into this at all.)